lib/connector/registration/create_organisation_converter.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:logotime/connector/view_model_converter.dart';
+import 'package:logotime/network/model/organisation/organisation_size.dart';
+import 'package:logotime/network/model/rules/rule_enums.dart';
+import 'package:logotime/network/model/rules/rules_model.dart';
+import 'package:logotime/redux/action/registration/create_organisation_action.dart';
+import 'package:logotime/widget/registration/create_organisation_step/create_organisation_view_model.dart';
+
+class CreateOrganisationConverter
+    implements ViewModelConverter<CreateOrganisationViewModel> {
+  final String? organisationNameError;
+  final String? organisationSizeError;
+  final String? maxApplicationsError;
+  RulesModel organisationRules;
+  Function(dynamic) dispatch;
+
+  CreateOrganisationConverter({
+    required this.organisationNameError,
+    required this.organisationSizeError,
+    required this.organisationRules,
+    required this.dispatch,
+    required this.maxApplicationsError,
+  });
+
+  @override
+  CreateOrganisationViewModel build() {
+    return CreateOrganisationViewModel((viewModel) => viewModel
+      ..organisationNameError = organisationNameError
+      ..organisationSizeError = organisationSizeError
+      ..organisationRules = organisationRules.toBuilder()
+      ..onSubstituteMeRuleChanged = _onSubstituteMeRuleChanged
+      ..onSwapShiftRuleChanged = _onSwapShiftRuleChanged
+      ..onCheckInRuleChanged = _onCheckInRuleChanged
+      ..onUnassignedShiftRuleChanged = _onUnassignedShiftRuleChanged
+      ..onCreateOrganisationButtonPressed = _onCreateOrganisation);
+  }
+
+  void _onSubstituteMeRuleChanged({bool? isAllowed, bool? needApproval}) {
+    isAllowed ??=
```

Comment:
The substituteMe rule is controlled by two switches. I do not save the state of the switches (only the state of the rule, which dictates the value of the switch). 

Let's imagine that the parameters are not optional. So to call this method, when one of the switches changes I need to either assume that the other switch is true/false, or do something like:
```dart 
onChange: (value) =>
        widget.viewModel.onSubstituteMeRuleChanged(
        isAllowed: value,
        needApproval: rules.substituteMeRule == SubstituteMeRule.allowedWithApproval) //logic is here

```
Assuming the value of the other switch does not sound very good, so I decided to check the state of the rule itself. To remove the logic from UI, I made the parameters optional so that the switch only sets the parameter, which it knows the value of.

P.S. I first went with assuming the value of another switch and forgot to update it, so the switch still looks like this:
```dart
onChange: (value) =>
       widget.viewModel.onSubstituteMeRuleChanged(
       isAllowed: value,
       needApproval: true, //this should be removed
    ),
```

Reply:
got you. ok, proceed with your solution as it explains your intentions.